### PR TITLE
last changes

### PR DESCRIPTION
**Pull request by vlykhvar**

### CODE REVIEW

**Review by gigbat — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1 +1,40 @@
...
+- log in
+
+`USER:`
+- View all available game
+- View all Stadium
+- View game match
+- View own order list
+- Find math by date
+- Make an order
+- add a ticket to his shopping card
```
Comment (KseniiaMakarova) [inline README.md]:
```suggestion
- View all available games
- View all stadiums
- View game matches
- View his own order history
- Find match by date
- Make an order
- Add a ticket to his shopping cart
```

File: README.md [hunk 2]
```diff
@@ -1 +1,40 @@
...
+- Java 8
+- Maven
+- Maven Checkstyle Plugin
+- Spring Security
+- Spring MVC
+- Hibernate
+---
+**Running the Project**
+1. Download and install last JDK
+2. Download and install the MySQL
```
Comment (KseniiaMakarova) [inline README.md:37]: what about Tomcat?

File: README.md [hunk 3]
```diff
@@ -1 +1,40 @@
...
+3. Clone the project from GitHub
+4. Open in an IDE.
+5. Configure file db.properties by according to your database settings
```
Comment (KseniiaMakarova) [inline README.md]:
```suggestion
5. Configure file db.properties according to your database settings.
```

File: README.md [hunk 4]
```diff
@@ -1 +1,40 @@
...
+6. Run the project!
```
Comment (KseniiaMakarova) [inline README.md]: is admin injected in your app? how can I test the app with ADMIN role?

File: src/main/java/com/dev/football/controller/CinemaHallController.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cinema-halls")
+public class CinemaHallController {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/controller/CinemaHallController.java]: should we rename this?

File: src/main/java/com/dev/football/controller/CinemaHallController.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Autowired
+    public CinemaHallController(StadiumMapper stadiumMapper,
+                                StadiumService stadiumService) {
+        this.stadiumMapper = stadiumMapper;
+        this.stadiumService = stadiumService;
+    }
+
+    @GetMapping
+    public List<StadiumResponseDto> getAllCinemaHall() {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/controller/CinemaHallController.java]: all the methods as well

File: src/main/java/com/dev/football/controller/MovieController.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movies")
+public class MovieController {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/controller/MovieController.java]: same

File: src/main/java/com/dev/football/controller/MovieSessionController.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-sessions")
+public class MovieSessionController {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/controller/MovieSessionController.java]: same

also change authorization settings in SecurityConfig

File: src/main/java/com/dev/football/controller/ShoppingCartController.java [hunk 9]
```diff
@@ -40,7 +40,7 @@ public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
 
     @PostMapping("/movie-sessions")
     public void addMovieSession(Authentication authentication, @RequestParam Long movieSessionId) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/controller/ShoppingCartController.java]: rename this as well

File: src/main/java/com/dev/football/service/impl/GameSessionServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+public class GameSessionServiceImpl implements GameSessionService {
+    private final GameSessionDao gameSessionDao;
+
+    @Autowired
+    public GameSessionServiceImpl(GameSessionDao gameSessionDao) {
+        this.gameSessionDao = gameSessionDao;
+    }
+
+    @Override
+    public List<GameSession> findAvailableSessions(Long movieId, LocalDate date) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/service/impl/GameSessionServiceImpl.java]: rename this

File: src/main/java/com/dev/football/service/impl/GameServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+        return gameDao.add(game);
+    }
+
+    @Override
+    public List<Game> getAll() {
+        return gameDao.getAll();
+    }
+
+    @Override
+    public Game findById(Long movieId) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/service/impl/GameServiceImpl.java]: same

File: src/main/java/com/dev/football/service/StadiumService.java [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.dev.football.model.Stadium;
+import java.util.List;
+
+public interface StadiumService {
+    Stadium add(Stadium stadium);
+
+    List<Stadium> getAll();
+
+    Stadium findById(Long cinemaHallId);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/service/StadiumService.java]: same

File: src/main/java/com/dev/football/service/GameSessionService.java [hunk 13]
```diff
@@ -0,0 +1,18 @@
+package com.dev.football.service;
+
+import com.dev.football.model.GameSession;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
+public interface GameSessionService {
+    List<GameSession> findAvailableSessions(Long movieId, LocalDate date);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/service/GameSessionService.java]: same

File: src/main/java/com/dev/football/service/GameService.java [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.dev.football.model.Game;
+import java.util.List;
+
+public interface GameService {
+    Game add(Game game);
+
+    List<Game> getAll();
+
+    Game findById(Long movieId);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/service/GameService.java]: same

File: src/main/java/com/dev/football/model/dto/GameSessionResponseDto.java [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+    private Long stadiumId;
+    private String stadiumDescription;
+
+    public String getGameSessionDescription() {
+        return stadiumDescription;
+    }
+
+    public void setGameSessionDescription(String cinemaHallDescription) {
+        this.stadiumDescription = cinemaHallDescription;
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/model/dto/GameSessionResponseDto.java]: what is going on?))

File: src/main/java/com/dev/football/model/dto/GameSessionRequestDto.java [hunk 16]
```diff
@@ -1,15 +1,15 @@
-package com.dev.theater.model.dto;
+package com.dev.football.model.dto;
 
 import jakarta.validation.constraints.Min;
 import jakarta.validation.constraints.NotNull;
 
-public class MovieSessionRequestDto {
+public class GameSessionRequestDto {
     @Min(1)
     private Long movieId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/model/dto/GameSessionRequestDto.java]: rename this

File: src/main/java/com/dev/football/model/Ticket.java [hunk 17]
```diff
@@ -24,12 +24,12 @@ public void setId(Long id) {
...
 
-    public MovieSession getMovieSession() {
-        return movieSession;
+    public GameSession getMovieSession() {
+        return gameSession;
     }
 
-    public void setMovieSession(MovieSession movieSession) {
-        this.movieSession = movieSession;
+    public void setMovieSession(GameSession gameSession) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/model/Ticket.java]: rename getters and setters

File: src/main/java/com/dev/football/mapper/StadiumMapper.java [hunk 18]
```diff
@@ -0,0 +1,24 @@
+package com.dev.football.mapper;
+
+import com.dev.football.model.Stadium;
+import com.dev.football.model.dto.StadiumRequestDto;
+import com.dev.football.model.dto.StadiumResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StadiumMapper {
+    public StadiumResponseDto cinemaHallToDto(Stadium stadium) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/mapper/StadiumMapper.java]: fix method names

File: src/main/java/com/dev/football/mapper/GameMapper.java [hunk 19]
```diff
@@ -0,0 +1,24 @@
+package com.dev.football.mapper;
+
+import com.dev.football.model.Game;
+import com.dev.football.model.dto.GameRequestDto;
+import com.dev.football.model.dto.GameResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GameMapper {
+    public GameResponseDto movieToDto(Game game) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/mapper/GameMapper.java]: same

File: src/main/java/com/dev/football/dao/impl/StadiumDaoImpl.java [hunk 20]
```diff
@@ -1,35 +1,35 @@
...
-            return session.createQuery("from CinemaHall", CinemaHall.class).getResultList();
+            return session.createQuery("from Stadium", Stadium.class).getResultList();
         } catch (Exception e) {
             throw new CrudException("Error getting all hall", e);
         }
     }
 
     @Override
-    public CinemaHall findById(Long cinemaHallId) {
+    public Stadium findById(Long cinemaHallId) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/dao/impl/StadiumDaoImpl.java]: rename this

File: src/main/java/com/dev/football/dao/impl/StadiumDaoImpl.java [hunk 21]
```diff
@@ -1,35 +1,35 @@
...
     }
 
     @Override
-    public List<CinemaHall> getAll() {
+    public List<Stadium> getAll() {
         try (Session session = sessionFactory.openSession()) {
-            return session.createQuery("from CinemaHall", CinemaHall.class).getResultList();
+            return session.createQuery("from Stadium", Stadium.class).getResultList();
         } catch (Exception e) {
             throw new CrudException("Error getting all hall", e);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/dao/impl/StadiumDaoImpl.java]: same

File: src/main/java/com/dev/football/dao/impl/GameSessionDaoImpl.java [hunk 22]
```diff
@@ -81,10 +81,10 @@ public boolean delete(Long id) {
     }
 
     @Override
-    public Optional<MovieSession> getById(Long movieSessionId) {
+    public Optional<GameSession> getById(Long movieSessionId) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/dao/impl/GameSessionDaoImpl.java]: same

File: src/main/java/com/dev/football/dao/impl/GameSessionDaoImpl.java [hunk 23]
```diff
@@ -34,21 +34,21 @@ public MovieSessionDaoImpl(SessionFactory sessionFactory) {
...
-            session.update(movieSession);
+            session.update(gameSession);
             transaction.commit();
-            return movieSession;
+            return gameSession;
         } catch (Exception e) {
             if (transaction != null) {
                 transaction.rollback();
             }
             throw new CrudException("Can't update moviesession: "
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/dao/impl/GameSessionDaoImpl.java]: same

File: src/main/java/com/dev/football/dao/impl/GameSessionDaoImpl.java [hunk 24]
```diff
@@ -14,18 +14,18 @@
...
+public class GameSessionDaoImpl extends DaoImpl<GameSession> implements GameSessionDao {
     @Autowired
-    public MovieSessionDaoImpl(SessionFactory sessionFactory) {
+    public GameSessionDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
-    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+    public List<GameSession> findAvailableSessions(Long movieId, LocalDate date) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/dao/impl/GameSessionDaoImpl.java]: same

File: src/main/java/com/dev/football/dao/impl/GameDaoImpl.java [hunk 25]
```diff
@@ -1,35 +1,35 @@
...
     }
 
     @Override
-    public List<Movie> getAll() {
+    public List<Game> getAll() {
         try (Session session = sessionFactory.openSession()) {
-            return session.createQuery("from Movie", Movie.class).getResultList();
+            return session.createQuery("from Game", Game.class).getResultList();
         } catch (Exception e) {
             throw new CrudException("Error getting all movies", e);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/dao/impl/GameDaoImpl.java]: same for all exception messages

File: src/main/java/com/dev/football/dao/StadiumDao.java [hunk 26]
```diff
@@ -0,0 +1,10 @@
+package com.dev.football.dao;
+
+import com.dev.football.model.Stadium;
+import java.util.List;
+
+public interface StadiumDao extends GenericDao<Stadium> {
+    List<Stadium> getAll();
+
+    Stadium findById(Long cinemaHallId);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/dao/StadiumDao.java]: same

File: src/main/java/com/dev/football/dao/GameSessionDao.java [hunk 27]
```diff
@@ -0,0 +1,16 @@
+package com.dev.football.dao;
+
+import com.dev.football.model.GameSession;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
+public interface GameSessionDao extends GenericDao<GameSession> {
+    List<GameSession> findAvailableSessions(Long movieId, LocalDate date);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/dao/GameSessionDao.java]: same

File: src/main/java/com/dev/football/dao/GameDao.java [hunk 28]
```diff
@@ -0,0 +1,10 @@
+package com.dev.football.dao;
+
+import com.dev.football.model.Game;
+import java.util.List;
+
+public interface GameDao extends GenericDao<Game> {
+    List<Game> getAll();
+
+    Game findById(Long movieId);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/dao/GameDao.java]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
remove all redundant branches from GitHub please

in your IDE - open global search in the project directory and search `movie` or `cinema`
all the occurrences should be replaced

File: src/main/java/com/dev/football/model/dto/GameSessionRequestDto.java (hunk 16 above)
Comment (vlykhvar) [reply src/main/java/com/dev/football/model/dto/GameSessionRequestDto.java]: strange. it was renamed

File: src/main/java/com/dev/football/mapper/StadiumMapper.java (hunk 18 above)
Comment (vlykhvar) [reply src/main/java/com/dev/football/mapper/StadiumMapper.java]: strange2. it was renamed

File: src/main/java/com/dev/football/mapper/GameMapper.java (hunk 19 above)
Comment (vlykhvar) [reply src/main/java/com/dev/football/mapper/GameMapper.java]: strange3. it was renamed

File: src/main/java/com/dev/football/dao/impl/StadiumDaoImpl.java (hunk 20 above)
Comment (vlykhvar) [reply src/main/java/com/dev/football/dao/impl/StadiumDaoImpl.java:29]: strange3. it was renamed

File: src/main/java/com/dev/football/model/dto/GameSessionResponseDto.java [hunk 29]
```diff
@@ -0,0 +1,49 @@
...
+    private String gameTitle;
+    private String showTime;
+    private Long stadiumId;
+    private String stadiumDescription;
+
+    public String getGameSessionDescription() {
+        return stadiumDescription;
+    }
+
+    public void setGameSessionDescription(String stadiumDescription) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/model/dto/GameSessionResponseDto.java]:
```suggestion
    public String getStadiumDescription() {
        return stadiumDescription;
    }

    public void setStadiumDescription(String stadiumDescription) {
```

File: src/main/java/com/dev/football/mapper/GameSessionMapper.java [hunk 30]
```diff
@@ -0,0 +1,41 @@
...
+        this.stadiumService = stadiumService;
+    }
+
+    public GameSessionResponseDto gameSessionToDto(GameSession gameSession) {
+        GameSessionResponseDto gameSessionDto = new GameSessionResponseDto();
+        gameSessionDto.setId(gameSession.getId());
+        gameSessionDto.setShowTime(gameSession.getShowTime().toString());
+        gameSessionDto.setGameTitle(gameSession.getGame().getTitle());
+        gameSessionDto.setStadiumId(gameSession.getStadium().getId());
+        gameSessionDto.setGameSessionDescription(gameSession.getStadium().getDescription());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/mapper/GameSessionMapper.java]:
```suggestion
        gameSessionDto.setStadiumDescription(gameSession.getStadium().getDescription());
```

File: src/main/java/com/dev/football/dao/impl/StadiumDaoImpl.java [hunk 31]
```diff
@@ -0,0 +1,39 @@
...
+    public StadiumDaoImpl(SessionFactory sessionFactory) {
+        super(sessionFactory);
+    }
+
+    @Override
+    public List<Stadium> getAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from Stadium", Stadium.class).getResultList();
+        } catch (Exception e) {
+            throw new CrudException("Error getting all hall", e);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/dao/impl/StadiumDaoImpl.java]:
```suggestion
            throw new CrudException("Error getting all stadiums", e);
```

File: src/main/java/com/dev/football/dao/impl/GameDaoImpl.java [hunk 32]
```diff
@@ -0,0 +1,39 @@
...
+    public GameDaoImpl(SessionFactory sessionFactory) {
+        super(sessionFactory);
+    }
+
+    @Override
+    public List<Game> getAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from Game", Game.class).getResultList();
+        } catch (Exception e) {
+            throw new CrudException("Error getting all movies", e);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/dao/impl/GameDaoImpl.java]:
```suggestion
            throw new CrudException("Error getting all games", e);
```

File: src/main/java/com/dev/football/dao/impl/GameDaoImpl.java [hunk 33]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Override
+    public Game findById(Long gameId) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from Game "
+                    + "where id = :gameId", Game.class)
+                    .setParameter("gameId", gameId)
+                    .getSingleResult();
+        } catch (Exception e) {
+            throw new CrudException("Can't get movie on id " + gameId);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/football/dao/impl/GameDaoImpl.java]:
```suggestion
            throw new CrudException("Can't get game on id " + gameId);
```

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**